scipy/stats/_survival.py

```diff
@@ -0,0 +1,116 @@
+from dataclasses import make_dataclass
+import numpy as np
+from ._censored_data import CensoredData
+
+
+__all__ = ['ecdf']
+
+
+ECDFResult = make_dataclass('ECDFResult', ['x', 'cdf', 'sf'])
+
+
+def ecdf(sample):
+    """Empirical cumulative distribution function of a sample
+
+    The empirical cumulative distribution function (ECDF) is a step function
+    estimate of the CDF of the distribution underlying a sample.
+
+    Parameters
+    ----------
+    sample : 1D array_like or `stats.CensoredData`
+        Besides array_like, instances of `stats.CensoredData` containing
+        uncensored observations are supported. Currently, instances of
+        `stats.CensoredData` with censored data will result in a
+        ``NotImplementedError``, but future support for left-censored,
+        right-centered, and interval-censored data is planned.
```

Comment:
What about just keeping this out for this PR? I expect we will have enough time to do the `stats.CensoredData` before the next release.

Reply:
Yes. So wouldn't in or out be OK?

Reply:
Fine with both, mostly saying this so we don't discuss it more than that during this review.